pkg/apis/node/v1alpha1/types.go

```diff
@@ -278,3 +282,42 @@ type CustomNodeResourceList struct {
 	// items is the list of CNRs
 	Items []CustomNodeResource `json:"items"`
 }
+
+// NodeMetricStatus defines the observed state of NodeMetric
+type NodeMetricStatus struct {
+	// UpdateTime is the last time this NodeMetricStatus was updated.
+	UpdateTime metav1.Time `json:"updateTime"`
+
+	// NodeMetric contains the metrics for this node.
+	NodeMetric *NodeMetricInfo `json:"nodeMetric,omitempty"`
+
+	// GroupMetric contains the metrics aggregated by QoS level groups
+	GroupMetric []GroupMetricInfo `json:"groupMetric,omitempty"`
+}
+
+type ResourceUsage struct {
+	// NUMAUsage contains the real-time resource usage for each NUMA
+	NUMAUsage []NUMAMetricInfo `json:"numaUsage,omitempty"`
+
+	// GenericUsage contains the real-time resource usage, in format of '{"cpu": 2, "memory": "1Gi"}'
+	GenericUsage *v1.ResourceList `json:"genericUsage,omitempty"`
+}
+
+type GroupMetricInfo struct {
```

Comment:
pls add comments to explain what is `Group` and clarify in which case we should use this concept; we don't want to make it too flexible such that it messes up

Reply:
Please refer to the comment of GroupMetric: GroupMetric contains the metrics aggregated by QoS level groups